File: Resources/CourseraAlgorithmicToolbox/week5_dynamic_programming1/5_longest_common_subsequence_of_three_sequences/by_learners/lcs3.rb
#!/usr/bin/env ruby
# by Andronik Ordian

def lcs3(a, b, c)
  # write your code here
  [a.size, b.size, c.size].min
end

if __FILE__ == $0
  data = STDIN.read.split().map(&:to_i)
  n = data[0]
  a = data[1..n]
  m = data[n+1]
  b = data[n+2..n+1+m]
  c = data[n+3+m..-1]
  puts lcs3(a, b, c)
end

File: Resources/CourseraAlgorithmicToolbox/week5_dynamic_programming1/5_longest_common_subsequence_of_three_sequences/LCS3.kt
import java.util.*
import kotlin.math.min

fun lcs3(a: IntArray, b: IntArray, c: IntArray): Int {
    //Write your code here
    return min(min(a.size, b.size), c.size)
}

fun main(args: Array<String>) {
    val scanner = Scanner(System.`in`)
    val an = scanner.nextInt()
    val a = IntArray(an)
    for (i in 0 until an) {
        a[i] = scanner.nextInt()
    }
    val bn = scanner.nextInt()
    val b = IntArray(bn)
    for (i in 0 until bn) {
        b[i] = scanner.nextInt()
    }
    val cn = scanner.nextInt()
    val c = IntArray(cn)
    for (i in 0 until cn) {
        c[i] = scanner.nextInt()
    }
    println(lcs3(a, b, c))
}